### DISCUSSION TITLE
Is `install` the same as `postinstall`?

### DISCUSSION THREAD

**Discussion by sachinraja** [npm/feedback#456] [CLI] [answered]
I'm trying to run a script after a contributor installs the library (no argument install). It seems like there's two options for this: `install` and `postinstall`. To me, it seems like the only difference is `postinstall` runs after `install`,   but in the docs it says this:
> Don't use install. Use a .gyp file for compilation, and prepublish for anything else. You should almost never have to explicitly set a preinstall or install script. If you are doing this, please consider if there is another option. The only valid use of install or preinstall scripts is for compilation which must be done on the target architecture.

[reference](https://docs.npmjs.com/cli/v7/using-npm/scripts#best-practices)

Does this also apply to `postinstall`?

**Comment by MylesBorins** [comment, marked as answer]
Hey @sachinraja FWIW this form is not meant for general technical support. For questions like this in the future it is better to use https://github.community or Stack Overflow

In this case you want to use `postinstall`. If you define the `install` script it will override `npm install` itself.

**Reply by sachinraja** [reply]
Hi, sorry for asking in the wrong place. Defining an `install` script does not override `npm install` though. The package is installed and the script just runs after installation.
